src/ape/api/address.py

```diff
@@ -20,6 +20,10 @@ def provider(self) -> ProviderAPI:
 
         return self._provider
 
+    @provider.setter
+    def provider(self, value: ProviderAPI):
+        self._provider = value
```

Comment:
Hmm... Wonder if we could just remove the `_provider` field, since we can inject it manually via `a = AddressAPI(); a.provider = ...`

Reply:
Or even more generally, make "injectable properties" which are properties that are injected after initialization :thinking:

Reply:
Interesting!

Reply:
Yeah, have to think of this one more. This property in particular is extremely awkward, and I couldn't figure out how to make it work properly. Same with the managers as properties (which is the source of the issue w/ pydantic refactor)

Reply:
this should be a new PR, not a blocker on this